presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotFilterExpressionConverter.java

```diff
@@ -0,0 +1,219 @@
+            RowExpression value = between.getArguments().get(0);
+            RowExpression min = between.getArguments().get(1);
+            RowExpression max = between.getArguments().get(2);
+
+            return derived(format("(%s BETWEEN %s AND %s)",
+                    value.accept(this, context).getDefinition(),
+                    min.accept(this, context).getDefinition(),
+                    max.accept(this, context).getDefinition()));
+        }
+
+        throw new PinotException(PINOT_UNSUPPORTED_EXPRESSION, Optional.empty(), format("Between operator not supported: %s", between));
+    }
+
+    private PinotExpression handleNot(CallExpression not, Function<VariableReferenceExpression, Selection> context)
+    {
+        if (not.getArguments().size() == 1) {
+            RowExpression input = not.getArguments().get(0);
+            if (input instanceof SpecialFormExpression) {
+                SpecialFormExpression specialFormExpression = (SpecialFormExpression) input;
+                // NOT operator is only supported on top of the IN expression
+                if (specialFormExpression.getForm() == SpecialFormExpression.Form.IN) {
+                    return handleIn(specialFormExpression, false, context);
+                }
+            }
+        }
+
+        throw new PinotException(PINOT_UNSUPPORTED_EXPRESSION, Optional.empty(), format("NOT operator is supported only on top of IN operator. Received: %s", not));
+    }
+
+    private PinotExpression handleCast(CallExpression cast, Function<VariableReferenceExpression, Selection> context)
+    {
+        if (cast.getArguments().size() == 1) {
+            RowExpression input = cast.getArguments().get(0);
+            Type expectedType = cast.getType();
+            if (typeManager.canCoerce(input.getType(), expectedType)) {
+                return input.accept(this, context);
+            }
+            else {
+                throw new PinotException(PINOT_UNSUPPORTED_EXPRESSION, Optional.empty(), "Non implicit casts not supported: " + cast);
+            }
+        }
+
+        throw new PinotException(PINOT_UNSUPPORTED_EXPRESSION, Optional.empty(), format("This type of CAST operator not supported. Received: %s", cast));
+    }
+
+    @Override
+    public PinotExpression visitCall(CallExpression call, Function<VariableReferenceExpression, Selection> context)
+    {
+        String displayName = call.getDisplayName().toLowerCase(ENGLISH);
+        if (displayName.equals("not")) {
+            return handleNot(call, context);
+        }
+        else if (displayName.equals("cast")) {
+            return handleCast(call, context);
+        }
+        else if (displayName.equals("between")) {
+            return handleBetween(call, context);
+        }
+        FunctionMetadata functionMetadata = functionMetadataManager.getFunctionMetadata(call.getFunctionHandle());
+        Optional<OperatorType> operatorTypeOptional = functionMetadata.getOperatorType();
```

Comment:
Instead of getting the operator type, just use `isArithmeticFunction` and `isArithmeticFunction` from `StandardFunctionResolution `